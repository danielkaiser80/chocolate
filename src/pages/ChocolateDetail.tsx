import { Box, Card, Typography, Link as MUILink } from "@mui/material";
import { findChocolate } from "../backend/chocolate.service.ts";
import Header from "../components/Header.tsx";
import { Link } from "wouter";
import PriceCard from "../components/PriceCard.tsx";
import { Fragment } from "react";

interface ChocolateDetailProps {
  id: string;
}

const ChocolateDetail = ({ id }: ChocolateDetailProps) => {
  const chocolate = findChocolate(id);
  if (!chocolate) {
    return <Card>Schokolade mit ID {id} wurde nicht gefunden.</Card>;
  }
  const { name, brand, prices, cheapestPricePer100g } = chocolate;

  return (
    <>
      <Header title={`Details: üç´${name}`} />

      <Box mb={2}>
        <Typography variant="h5">Hersteller: {brand}</Typography>
      </Box>

      {prices.length > 0 ? (
        prices.map((price) => (
          <Fragment key={price.shop + price.price}>
            <PriceCard
              price={price}
              cheapestPricePer100g={cheapestPricePer100g}
            />
            <hr />
          </Fragment>
        ))
      ) : (
        <Box>
          <Typography variant="body2">Derzeit nicht verf√ºgbar üò≠</Typography>
        </Box>
      )}

      <Typography variant="body1">
        Der g√ºnstigste Preis je 100 Gramm ist rot hervorgehoben!
      </Typography>

      <Box>
        <Link href="/">
          <MUILink>Zur√ºck zur √úbersicht</MUILink>
        </Link>
      </Box>
    </>
  );
};

export default ChocolateDetail;
