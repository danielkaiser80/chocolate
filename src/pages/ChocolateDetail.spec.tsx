import { render } from "@testing-library/react";
import ChocolateDetail from "./ChocolateDetail.tsx";

describe(ChocolateDetail, () => {
  it("should render the detail view correctly", () => {
    const result = render(<ChocolateDetail id="1" />);
    expect(result).toMatchSnapshot();
  });

  it("should render the detail view correctly for not found", () => {
    const result = render(<ChocolateDetail id="XYZ" />);
    expect(result).toMatchSnapshot();
  });
});
