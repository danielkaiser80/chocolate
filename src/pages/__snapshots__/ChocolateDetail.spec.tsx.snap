// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChocolateDetail > should render the detail view correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-12waxkz"
      >
        <h1
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        >
          Details: üç´Sesam
        </h1>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Entdecken Sie eine Vielfalt an k√∂stlichen Schokoladen-Tafeln von Spitzenmarken.
        </p>
        <img
          alt="Chocolate Bar"
          src="https://www.greenamerica.org/sites/default/files/2020-04/iStock-893183000%20-%20stacked%20chocolate%20bars.jpg"
          style="max-width: 200px;"
        />
      </div>
      <div
        class="MuiBox-root css-1qm1lh"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          Hersteller: 
          Ritter-Sport
        </h5>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiBox-root css-1xlzx9v"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Preis: 
              2,09
               ‚Ç¨
            </h5>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Anbieter: 
              Ritter-Sport
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Menge: 
              100
               
              g
            </p>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
              href="https://example.com/rs-sesam-rs"
            >
              Jetzt kaufen
            </a>
          </div>
        </div>
      </div>
      <hr />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiBox-root css-1xlzx9v"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Preis: 
              1,99
               ‚Ç¨
            </h5>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Anbieter: 
              Amazon
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Menge: 
              100
               
              g
            </p>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
              href="https://example.com/rs-sesam-am"
            >
              Jetzt kaufen
            </a>
          </div>
        </div>
      </div>
      <hr />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiBox-root css-1xlzx9v"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-8zamrl-MuiTypography-root"
            >
              Preis: 
              3,49
               ‚Ç¨
            </h5>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Anbieter: 
              REWE
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Menge: 
              200
               
              g
            </p>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
              href="https://example.com/rs-sesam-re"
            >
              Jetzt kaufen
            </a>
          </div>
        </div>
      </div>
      <hr />
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Der g√ºnstigste Preis je 100 Gramm ist rot hervorgehoben!
      </p>
      <div
        class="MuiBox-root css-0"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
          href="/"
        >
          Zur√ºck zur √úbersicht
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-12waxkz"
    >
      <h1
        class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
      >
        Details: üç´Sesam
      </h1>
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Entdecken Sie eine Vielfalt an k√∂stlichen Schokoladen-Tafeln von Spitzenmarken.
      </p>
      <img
        alt="Chocolate Bar"
        src="https://www.greenamerica.org/sites/default/files/2020-04/iStock-893183000%20-%20stacked%20chocolate%20bars.jpg"
        style="max-width: 200px;"
      />
    </div>
    <div
      class="MuiBox-root css-1qm1lh"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
      >
        Hersteller: 
        Ritter-Sport
      </h5>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          class="MuiBox-root css-1xlzx9v"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          >
            Preis: 
            2,09
             ‚Ç¨
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Anbieter: 
            Ritter-Sport
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Menge: 
            100
             
            g
          </p>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
            href="https://example.com/rs-sesam-rs"
          >
            Jetzt kaufen
          </a>
        </div>
      </div>
    </div>
    <hr />
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          class="MuiBox-root css-1xlzx9v"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          >
            Preis: 
            1,99
             ‚Ç¨
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Anbieter: 
            Amazon
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Menge: 
            100
             
            g
          </p>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
            href="https://example.com/rs-sesam-am"
          >
            Jetzt kaufen
          </a>
        </div>
      </div>
    </div>
    <hr />
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          class="MuiBox-root css-1xlzx9v"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-8zamrl-MuiTypography-root"
          >
            Preis: 
            3,49
             ‚Ç¨
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Anbieter: 
            REWE
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Menge: 
            200
             
            g
          </p>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
            href="https://example.com/rs-sesam-re"
          >
            Jetzt kaufen
          </a>
        </div>
      </div>
    </div>
    <hr />
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Der g√ºnstigste Preis je 100 Gramm ist rot hervorgehoben!
    </p>
    <div
      class="MuiBox-root css-0"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
        href="/"
      >
        Zur√ºck zur √úbersicht
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ChocolateDetail > should render the detail view correctly for not found 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        Schokolade mit ID 
        XYZ
         wurde nicht gefunden.
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      Schokolade mit ID 
      XYZ
       wurde nicht gefunden.
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
